VERSION ""


NS_ :
    NS_DESC_
    CM_
    BA_DEF_
    BA_
    VAL_
    CAT_DEF_
    CAT_
    FILTER
    BA_DEF_DEF_
    EV_DATA_
    ENVVAR_DATA_
    SGTYPE_
    SGTYPE_VAL_
    BA_DEF_SGTYPE_
    BA_SGTYPE_
    SIG_TYPE_REF_
    VAL_TABLE_
    SIG_GROUP_
    SIG_VALTYPE_
    SIGTYPE_VALTYPE_
    BO_TX_BU_
    BA_DEF_REL_
    BA_REL_
    BA_DEF_DEF_REL_
    BU_SG_REL_
    BU_EV_REL_
    BU_BO_REL_
    SG_MUL_VAL_

BS_:

BU_: XXX
BO_ 62 ENGINE_DATA: 48 XXX
 SG_ NEW_SIGNAL_4 : 17|10@0+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_5 : 18|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 23|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_3 : 38|15@0- (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_2 : 55|1@0+ (1,0) [0|1] "" XXX
 SG_ GAS_POS : 85|14@0+ (1,-600) [0|127] "" XXX
 SG_ NEW_SIGNAL_7 : 102|15@0+ (1,0) [0|32767] "" XXX
 SG_ NEW_SIGNAL_9 : 103|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_6 : 118|15@0+ (1,0) [0|32767] "" XXX
 SG_ NEW_SIGNAL_13 : 119|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_15 : 151|16@0+ (1,0) [0|65535] "" XXX
 SG_ NEW_SIGNAL_11 : 166|7@0+ (1,0) [0|127] "" XXX
 SG_ NEW_SIGNAL_12 : 174|7@0+ (1,0) [0|127] "" XXX
 SG_ GAS : 183|16@0- (1,0) [0|32767] "" XXX
 SG_ GEAR : 208|3@1+ (1,0) [0|63] "" XXX
 SG_ GEAR_BUTTON : 211|3@1+ (1,0) [0|7] "" XXX
 SG_ ACC_STATE : 218|3@0+ (1,0) [0|7] "" XXX
 SG_ BRAKE_PRESS : 220|1@0+ (1,0) [0|1] "" XXX
 SG_ CRUIZE_SPEED : 229|6@0+ (1,0) [0|63] "" XXX
 SG_ NEW_SIGNAL_16 : 234|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_10 : 239|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_14 : 278|15@0+ (1,0) [0|32767] "" XXX

BO_ 196 STEER_SENSOR: 8 XXX
 SG_ ANGLE : 3|8@0+ (1,0) [0|32767] "" XXX
 SG_ FRACTION : 4|4@1+ (1,0) [0|15] "" XXX
 SG_ TORQUE : 8|4@1- (1,0) [0|15] "" XXX
 SG_ NEW_SIGNAL_2 : 16|6@1- (1,0) [0|127] "" XXX
 SG_ NEW_SIGNAL_3 : 25|3@1+ (1,0) [0|3] "" XXX

BO_ 295 NEW_MSG_127: 8 XXX
 SG_ NEW_SIGNAL_2 : 24|6@1+ (1,0) [0|63] "" XXX
 SG_ NEW_SIGNAL_3 : 30|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 39|8@0- (1,0) [0|65535] "" XXX

BO_ 296 NEW_MSG_128: 8 XXX
 SG_ NEW_SIGNAL_1 : 19|12@0+ (1,0) [0|4095] "" XXX

BO_ 467 STEER_ANGLE_SENSOR: 8 XXX
 SG_ STEER_ANGLE : 6|13@0- (1,390) [0|16383] "" XXX
 SG_ NEW_SIGNAL_1 : 7|1@0+ (1,0) [0|1] "" XXX
 SG_ STEER_TORQUE_DRIVER : 16|6@1- (1,0) [0|31] "" XXX
 SG_ COUNTER : 48|5@1+ (1,0) [0|31] "" XXX
 SG_ CHECKSUM : 56|8@1+ (1,0) [0|255] "" XXX

BO_ 483 LKAS: 8 XXX
 SG_ NEW_SIGNAL_1 : 0|3@1+ (1,0) [0|7] "" XXX
 SG_ NEW_SIGNAL_7 : 3|5@1- (1,0) [0|31] "" XXX
 SG_ LKAS_CMD : 14|10@0- (1,0) [0|2047] "" XXX
 SG_ ACTIVE : 15|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_5 : 20|5@0+ (1,0) [0|31] "" XXX
 SG_ NEW_SIGNAL_2 : 24|8@1+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_3 : 32|8@1+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_4 : 40|8@1+ (1,0) [0|255] "" XXX
 SG_ COUNTER : 48|8@1+ (1,0) [0|15] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|4095] "" XXX

BO_ 608 NEW_MSG_260: 8 XXX
 SG_ NEW_SIGNAL_3 : 2|10@0+ (1,0) [0|1023] "" XXX
 SG_ BRAKE_PRESS : 16|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_4 : 31|7@0+ (1,0) [0|127] "" XXX
 SG_ NEW_SIGNAL_2 : 32|1@0+ (1,0) [0|65535] "" XXX
 SG_ NEW_SIGNAL_1 : 47|12@0- (1,0) [0|4095] "" XXX

BO_ 666 BRAKE_DATA: 8 XXX
 SG_ NEW_SIGNAL_3 : 3|4@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_2 : 15|2@0+ (1,0) [0|1] "" XXX
 SG_ BRAKE_POS : 21|12@0- (1,-100) [0|511] "" XXX

BO_ 745 NEW_MSG_2E9: 8 XXX

BO_ 773 NEW_MSG_305: 8 XXX
 SG_ NEW_SIGNAL_2 : 48|4@1+ (1,0) [0|15] "" XXX
 SG_ NEW_SIGNAL_1 : 63|8@0+ (1,0) [0|255] "" XXX

BO_ 775 LKAS_STATE: 8 XXX
 SG_ NEW_SIGNAL_3 : 6|2@1+ (1,0) [0|3] "" XXX
 SG_ NEW_SIGNAL_2 : 14|2@1+ (1,0) [0|3] "" XXX
 SG_ STATE : 45|1@0+ (1,0) [0|1] "" XXX
 SG_ LKA_ACTIVE : 46|1@1+ (1,0) [0|3] "" XXX
 SG_ COUNTER : 48|4@1+ (1,0) [0|15] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX

BO_ 787 WHEEL_SPEED_REAR: 8 XXX
 SG_ WHEEL_SPEED_RR : 6|15@0+ (0.00828,0) [0|32767] "" XXX
 SG_ WHEEL_SPEED_RL : 22|15@0+ (0.00828,0) [0|32767] "" XXX
 SG_ NEW_SIGNAL_1 : 39|8@0+ (1,0) [0|63] "" XXX

BO_ 790 WHEEL_SPEED_FRNT: 8 XXX
 SG_ WHEEL_SPEED_FR : 6|15@0- (0.00829,0) [-2000|32767] "kmh" XXX
 SG_ WHEEL_SPEED_FL : 22|15@0- (0.00829,0) [0|32767] "" XXX
 SG_ NEW_SIGNAL_1 : 32|7@1+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_2 : 39|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_3 : 41|2@0+ (1,0) [0|3] "" XXX

BO_ 793 NEW_MSG_319: 8 XXX

BO_ 837 LKAS_CAM_CMD_345: 8 XXX
 SG_ CMD : 6|13@0- (1,0) [0|8191] "" XXX
 SG_ NEW_SIGNAL_3 : 7|1@0+ (1,0) [0|1] "" XXX
 SG_ LKA_ACTIVE : 9|1@0+ (1,0) [0|1] "" XXX
 SG_ SET_X0 : 16|8@1+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_5 : 31|8@0+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_6 : 39|8@0+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_7 : 47|8@0+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_1 : 55|8@0+ (1,0) [0|255] "" XXX
 SG_ CHECKSUM : 56|8@1+ (1,0) [0|255] "" XXX

BO_ 864 STEER_BUTTON: 6 XXX
 SG_ CHECKSUM : 0|8@1+ (1,0) [0|255] "" XXX
 SG_ COUNTER : 12|4@1+ (1,0) [0|15] "" XXX
 SG_ ACC : 24|1@0+ (1,0) [0|1] "" XXX
 SG_ CC_BTN : 26|1@0+ (1,0) [0|1] "" XXX
 SG_ RES_PLUS : 30|1@0+ (1,0) [0|1] "" XXX
 SG_ RES_MINUS : 32|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 38|6@0+ (1,0) [0|1] "" XXX
 SG_ GAP_ADJUST_DOWN : 43|1@0+ (1,0) [0|1] "" XXX
 SG_ GAP_ADJUST_UP : 45|1@0+ (1,0) [0|1] "" XXX

BO_ 903 SETTING: 8 XXX
 SG_ NEW_SIGNAL_6 : 1|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_4 : 4|1@0+ (1,0) [0|1] "" XXX
 SG_ GAP : 11|3@0+ (1,0) [0|7] "" XXX
 SG_ NEW_SIGNAL_3 : 29|6@0+ (1,0) [0|31] "" XXX
 SG_ NEW_SIGNAL_2 : 34|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 35|3@1+ (1,0) [0|15] "" XXX
 SG_ CC_SPEED : 43|7@0+ (1,0) [0|1023] "" XXX
 SG_ COUNTER : 51|4@0+ (1,0) [0|15] "" XXX
 SG_ CHECKSUM : 63|8@0+ (1,0) [0|255] "" XXX

BO_ 913 BCM_SIGNAL_2: 8 XXX
 SG_ RIGHT_SIGN : 0|1@0+ (1,0) [0|1] "" XXX
 SG_ LEFT_SIGN : 1|1@0+ (1,0) [0|1] "" XXX
 SG_ HIGH_BEAM : 2|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_6 : 3|5@1+ (1,0) [0|31] "" XXX
 SG_ NEW_SIGNAL_5 : 15|2@0+ (1,0) [0|63] "" XXX
 SG_ NEW_SIGNAL_8 : 17|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_4 : 23|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_3 : 35|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_7 : 44|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 46|2@1+ (1,0) [0|3] "" XXX
 SG_ NEW_SIGNAL_2 : 53|1@0+ (1,0) [0|1] "" XXX

BO_ 914 BCM_SIGNAL_1: 8 XXX
 SG_ SIGN_SIGNAL : 16|2@1+ (1,0) [0|3] "" XXX
 SG_ NEW_SIGNAL_2 : 31|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_3 : 35|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 47|3@0+ (1,0) [0|7] "" XXX

BO_ 916 STEER_SENSOR_2: 8 XXX
 SG_ TORQUE_DRIVER : 6|11@0- (0.24,0) [0|1023] "" XXX
 SG_ NEW_SIGNAL_2 : 7|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 35|1@0+ (1,0) [0|1] "" XXX

BO_ 930 ACC_CMD: 8 XXX
 SG_ CMD : 6|10@0- (1,0) [0|127] "" XXX
 SG_ ACCEL_ON : 7|1@0+ (1,0) [0|1] "" XXX
 SG_ ACC_STATE : 9|2@0+ (1,0) [0|1] "" XXX
 SG_ STOPPED : 10|1@0+ (1,0) [0|1] "" XXX
 SG_ ACC_STATE_2 : 11|2@1+ (1,0) [0|15] "" XXX
 SG_ NEW_SIGNAL_12 : 16|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_9 : 17|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_2 : 22|5@0+ (1,0) [0|31] "" XXX
 SG_ STOPPING : 23|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_13 : 24|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_8 : 25|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_5 : 30|5@0+ (1,0) [0|31] "" XXX
 SG_ NEW_SIGNAL_6 : 31|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_10 : 32|7@1+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_3 : 39|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_4 : 40|7@1- (1,0) [0|255] "" XXX
 SG_ GAS_PRESSED : 47|1@0+ (1,0) [0|1] "" XXX
 SG_ COUNTER : 48|4@1+ (1,0) [0|15] "" XXX
 SG_ NEW_SIGNAL_11 : 55|4@0+ (1,0) [0|15] "" XXX
 SG_ CHECKSUM : 56|8@1+ (1,0) [0|255] "" XXX

BO_ 933 ACC: 8 XXX
 SG_ ACC_ACTIVE : 20|1@0+ (1,0) [0|1] "" XXX
 SG_ GAS_PRESSED : 21|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_4 : 22|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_2 : 44|1@0+ (1,0) [0|1] "" XXX
 SG_ COUNTER : 48|4@1+ (1,0) [0|15] "" XXX
 SG_ CHECKSUM : 56|8@1+ (1,0) [0|255] "" XXX

BO_ 935 NEW_MSG_3A7: 24 XXX

BO_ 943 NEW_MSG_3AF: 8 XXX
 SG_ NEW_SIGNAL_3 : 5|14@0+ (1,0) [0|16383] "" XXX
 SG_ NEW_SIGNAL_1 : 22|14@0+ (1,0) [0|16383] "" XXX
 SG_ NEW_SIGNAL_5 : 23|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_2 : 37|2@1+ (1,0) [0|3] "" XXX
 SG_ CHECKSUM : 51|12@0+ (1,0) [0|4095] "" XXX

BO_ 988 NEW_MSG_3DC: 8 XXX
 SG_ NEW_SIGNAL_6 : 7|8@0+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_4 : 9|10@0+ (1,0) [0|8191] "" XXX
 SG_ NEW_SIGNAL_5 : 10|6@1+ (1,0) [0|63] "" XXX
 SG_ NEW_SIGNAL_1 : 31|8@0+ (1,0) [0|8191] "" XXX
 SG_ NEW_SIGNAL_2 : 33|10@0+ (1,0) [0|65535] "" XXX
 SG_ NEW_SIGNAL_3 : 34|6@1+ (1,0) [0|63] "" XXX
 SG_ COUNTER : 48|4@1+ (1,0) [0|15] "" XXX
 SG_ CHECKSUM : 56|8@1+ (1,0) [0|255] "" XXX

BO_ 990 NEW_MSG_3DE: 8 XXX
 SG_ NEW_SIGNAL_5 : 6|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 24|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_3 : 34|3@1+ (1,0) [0|7] "" XXX
 SG_ NEW_SIGNAL_2 : 40|8@1- (1,0) [0|31] "" XXX

BO_ 994 NEW_MSG_3E2: 8 XXX
 SG_ NEW_SIGNAL_2 : 7|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 21|10@0- (1,0) [0|1023] "" XXX

BO_ 1005 NEW_MSG_3ED: 8 XXX
 SG_ NEW_SIGNAL_2 : 48|4@1+ (1,0) [0|15] "" XXX
 SG_ NEW_SIGNAL_1 : 56|8@1+ (1,0) [0|255] "" XXX

BO_ 1018 NEW_MSG_3FA: 8 XXX
 SG_ NEW_SIGNAL_7 : 5|14@0+ (1,0) [0|2047] "" XXX
 SG_ NEW_SIGNAL_6 : 7|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_5 : 23|8@0+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_4 : 28|13@0+ (1,0) [0|4095] "" XXX
 SG_ NEW_SIGNAL_3 : 31|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_2 : 40|8@1+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_1 : 51|12@0+ (1,0) [0|4095] "" XXX

BO_ 1020 HUD_ALERT: 8 XXX
 SG_ NEW_SIGNAL_6 : 16|2@1+ (1,0) [0|3] "" XXX
 SG_ NEW_SIGNAL_7 : 19|2@1+ (1,0) [0|3] "" XXX
 SG_ ICA_WARNING : 25|3@1+ (1,0) [0|7] "" XXX
 SG_ NEW_SIGNAL_4 : 29|2@1+ (1,0) [0|3] "" XXX
 SG_ STEER_WARNING : 38|1@0+ (1,0) [0|1] "" XXX
 SG_ TAKE_OVER_WARNING : 40|5@1+ (1,0) [0|31] "" XXX
 SG_ NEW_SIGNAL_3 : 55|8@0- (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_5 : 63|8@0+ (1,0) [0|255] "" XXX

BO_ 1039 NEW_MSG_40F: 8 XXX
 SG_ NEW_SIGNAL_1 : 6|2@1+ (1,0) [0|3] "" XXX
 SG_ NEW_SIGNAL_2 : 8|2@1+ (1,0) [0|3] "" XXX
 SG_ NEW_SIGNAL_4 : 21|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_5 : 27|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_3 : 32|2@1+ (1,0) [0|3] "" XXX

BO_ 1063 NEW_MSG_427: 8 XXX
 SG_ NEW_SIGNAL_1 : 4|5@0+ (1,0) [0|31] "" XXX

BO_ 1072 NEW_MSG_430: 8 XXX
 SG_ NEW_SIGNAL_1 : 0|3@1+ (1,0) [0|3] "" XXX
 SG_ NEW_SIGNAL_2 : 12|4@1+ (1,0) [0|15] "" XXX
 SG_ NEW_SIGNAL_8 : 16|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_6 : 17|7@1+ (1,0) [0|127] "" XXX
 SG_ NEW_SIGNAL_5 : 24|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_7 : 27|3@0+ (1,0) [0|7] "" XXX
 SG_ NEW_SIGNAL_4 : 48|8@1+ (1,0) [0|255] "" XXX
 SG_ NEW_SIGNAL_3 : 56|8@1+ (1,0) [0|255] "" XXX

BO_ 1077 NEW_MSG_435: 8 XXX
 SG_ NEW_SIGNAL_1 : 23|8@0+ (1,0) [0|255] "" XXX

BO_ 1201 BSM_LEFT: 8 XXX
 SG_ BSM_LEFT_DETECT : 8|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 14|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_3 : 15|1@0+ (1,0) [0|1] "" XXX

BO_ 1203 BSM_RIGHT: 8 XXX
 SG_ BSM_RIGHT_DETECT : 4|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_1 : 14|1@0+ (1,0) [0|1] "" XXX

BO_ 1245 NEW_MSG_4DD: 8 XXX
 SG_ NEW_SIGNAL_2 : 48|4@1+ (1,0) [0|15] "" XXX
 SG_ NEW_SIGNAL_1 : 63|8@0+ (1,0) [0|255] "" XXX

BO_ 1251 NEW_MSG_4E3: 8 XXX
 SG_ NEW_SIGNAL_1 : 5|1@0+ (1,0) [0|1] "" XXX

BO_ 1261 BRAKE_SENSOR: 8 XXX
 SG_ NEW_SIGNAL_1 : 9|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_2 : 11|1@0+ (1,0) [0|1] "" XXX
 SG_ BRAKE_PRESS : 37|1@0+ (1,0) [0|1] "" XXX
 SG_ BRAKE_REQUEST : 40|1@0+ (1,0) [0|1] "" XXX
 SG_ NEW_SIGNAL_4 : 42|1@0+ (1,0) [0|1] "" XXX

BO_ 1333 NEW_MSG_535: 8 XXX

VAL_ 62 GEAR 1 "P" 2 "R" 3 "N" 4 "D";
VAL_ 914 SIGN_SIGNAL 1 "RIGHT" 2 "LEFT";
